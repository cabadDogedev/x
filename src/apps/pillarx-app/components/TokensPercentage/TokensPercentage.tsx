// components
import BodySmall from '../Typography/BodySmall';

// images
import TriangleGreenIcon from '../../images/triangle-up-green.svg';
import TriangleRedIcon from '../../images/triangle-down-red.svg';

type TokensPercentageProps = {
    percentage?: number | null;
}

const TokensPercentage = ({percentage}: TokensPercentageProps) => {
    if (!percentage) {
        return null;
    }

const trianglePercentage = percentage >= 0 ? TriangleGreenIcon : TriangleRedIcon;

    return (
        <div className='flex gap-1 items-center'>
            <img src={trianglePercentage} className='w-[10px] h[9px]' />
            <BodySmall className={`${percentage >= 0 ? 'text-percentage_green' : 'text-percentage_red'}`}>{percentage.toFixed(2)}%</BodySmall>
        </div>
    )
}

export default TokensPercentage;
