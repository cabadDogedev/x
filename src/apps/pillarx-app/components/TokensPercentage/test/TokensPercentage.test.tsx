import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

// components
import TokensPercentage from '../TokensPercentage';
import TriangleGreenIcon from '../../../images/triangle-up-green.svg';
import TriangleRedIcon from '../../../images/triangle-down-red.svg';


describe('<TokensPercentage />', () => {
    const percentage = 12.34;
  
    it('renders correctly and matches snapshot', () => {
      const tree = renderer
        .create(<TokensPercentage percentage={percentage} />)
        .toJSON();
  
      expect(tree).toMatchSnapshot();
    });
  
    it('renders the correct icon and number for positive percentage', () => {
      const tree = renderer.create(<TokensPercentage percentage={percentage} />).toJSON() as ReactTestRendererJSON;
  
      const percentageIcon = (tree.children?.find(child =>
        typeof child === 'object' && child.type === 'img'
      ) as ReactTestRendererJSON);
  
      const percentageText = (tree.children?.find(child =>
        typeof child === 'object' && child.type === 'p'
      ) as ReactTestRendererJSON);
  
      expect(percentageIcon).not.toBeNull();
      expect(percentageIcon.props.src).toBe(TriangleGreenIcon);
  
      expect(percentageText).not.toBeNull();
      expect(percentageText?.children?.[0]).toBe('12.34');
      expect(percentageText.props.className).toContain('text-percentage_green');
    });
  
    it('renders the correct icon and number for negative percentage', () => {
      const tree = renderer.create(<TokensPercentage percentage={-percentage} />).toJSON() as ReactTestRendererJSON;
  
      const percentageIcon = (tree.children?.find(child =>
        typeof child === 'object' && child.type === 'img'
      ) as ReactTestRendererJSON) || null;
  
      const percentageText = (tree.children?.find(child =>
        typeof child === 'object' && child.type === 'p'
      ) as ReactTestRendererJSON) || null;
  
      expect(percentageIcon).not.toBeNull();
      expect(percentageIcon.props.src).toBe(TriangleRedIcon);
  
      expect(percentageText).not.toBeNull();
      expect(percentageText?.children?.[0]).toBe('-12.34');
      expect(percentageText.props.className).toContain('text-percentage_red');
    });

    it('returns null when percentage is not provided', () => {
      const tree = renderer.create(<TokensPercentage />).toJSON();
      expect(tree).toBeNull();
  });

  it('returns null when percentage is null', () => {
      const tree = renderer.create(<TokensPercentage percentage={null} />).toJSON();
      expect(tree).toBeNull();
  });
});