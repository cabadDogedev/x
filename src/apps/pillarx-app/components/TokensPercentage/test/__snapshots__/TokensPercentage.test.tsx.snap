// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokensPercentage /> renders correctly and matches snapshot 1`] = `
<div
  className="flex gap-1 items-center"
>
  <img
    className="w-[10px] h[9px]"
    src={
      {
        "ReactComponent": {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "default": "triangle-up-green.svg",
      }
    }
  />
  <p
    className="text-sm font-medium text-percentage_green"
  >
    12.34
    %
  </p>
</div>
`;
